### ISSUE TITLE
lax.conv and numpy.convolve 

### ISSUE THREAD

**Issue by jessebett** [google/jax#1561]
`lax.conv` padding argument currently only takes `SAME` and `VALID`. 

Can it also support `FULL` like [numpy's `convolve`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.convolve.html)?

**Comment by jessebett** [comment]
Currently `jax.numpy.convolve` raises an implementation error. Would it be possible to call the `lax.conv` method for this?

There are some annoying mismatches between what `numpy` expects and what `lax.conv` expects.

Here's a couple:
* `lax.conv` expects `n+2` dimensional input array. (btw, this is pretty annoyingly written. The docstring should be `2+n` or explicitly say the convention is `(batches, channels, *window_dims)`)
`numpy.convolve` expects 1D arrays.
* The `rhs` argument needs to be flipped along the window axis to match `numpy.convolve` convention.
* `"SAME"`, currently not supported for padding in `lax.conv` is the default mode for `numpy.convolve`.

Here's an example:
```python
from jax import lax
import jax.numpy as np
import numpy as onp

## Numpy
x = np.array([1.,2.,3.])
y = np.array([4.,5.,6.])

onp.convolve(x,y) #padding mode "full"
#> array([ 4., 13., 28., 27., 18.], dtype=float32)

onp.convolve(x,y,"valid")
#> array([28.], dtype=float32)

onp.convolve(x,y,"same")
#> array([13., 28., 27.], dtype=float32)

# lax expects n+2 dim arrays
x_jax = np.reshape(x,(1,1,len(x)))
y_jax = np.flip(np.reshape(y,(1,1,len(y))),2) #flip rhs

# lax should support "full"
lax.conv_general_dilated(x_jax,y_jax,[1],[(len(x)-1,len(x)-1)])
#> DeviceArray([[[ 4., 13., 28., 27., 18.]]], dtype=float32)

lax.conv(x_jax,y_jax,(1,),"valid")
#> DeviceArray([[[28.]]], dtype=float32)

lax.conv(x_jax,y_jax,(1,),"same")
#>  DeviceArray([[[13., 28., 27.]]], dtype=float32)
```

**Comment by hawkinsp** [comment]
Yes, you're on the right track. Perhaps you would like to contribute a PR that implements `np.convolve` in terms of `lax.conv_general_dilated`?

You have essentially the correct implementation there, although I would phrase it in terms of the operations in `jax.lax` rather than the operations in `jax.numpy` if I were implementing it in `jax.numpy`. That's really all there is to it, other than writing a test.

As you note `lax.conv_general_dilated` supports a general integer padding specification which will let you implement any of SAME, VALID, or FULL.

**Comment by hawkinsp** [comment]
`jax.numpy.convolve` was recently added to `jax`, which resolves this issue.
